src/main/java/com/auctionapp/api/repository/specification/SpecificationFactory.java

```diff
@@ -0,0 +1,28 @@
+package com.auctionapp.api.repository.specification;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import com.auctionapp.api.model.entities.Category;
+
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+public class SpecificationFactory<T> {
```

Comment:
~~ The method names here seem too generic for the specific purpose they fulfill. Consider renaming them to something more descriptive, or making them more generic in their implementation.